src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1247 @@
+                    PRINT_DEBUG(
+                        "[HYPERBEE] Failed to add execute trap X on GFN 0x%lx. Deleting mmAccessFault Return Trap\n",
+                        info->trap->memaccess.gfn);
+                }
+            }
+            else
+            {
+                PRINT_DEBUG(
+                    "[HYPERBEE] Failed to create execute trap X. Not monitoring GFN 0x%lx\n",
+                    info->trap->memaccess.gfn);
+            }
+        }
+        else
+        {
+            PRINT_DEBUG(
+                "[HYPERBEE] Failed to allocate memory for fault_data. Not monitoring GFN 0x%lx\n",
+                info->trap->memaccess.gfn);
+        }
+
+    }
+
+    //Destroys this return trap, because it is specific for the RIP and not usable anymore. This was the trap being called when the physical address got computed.
+    //Deletes this trap from the list of existing traps traps
+    //Additionally removes the trap and frees the memory
+    plugin->destroy_trap(info->trap);
+
+    return VMI_EVENT_RESPONSE_NONE;
+}
+
+/**
+ * This method is the callback of a trap which was added in the last lines of the constructor through register_trap
+ * It is called at the entry of MmAccessFault right before the first instruction of the access fault handler is executed.
+ * At that point we just know the faulting virtual address, which is the second parameter -> rdx.
+ * At that moment the fault handler was still not executed, so this Virtual Address is still not mapped to anywhere. Thus we create another breakpoint, which causes that mm_access_fault_return_hook_cb will be called one MmAccessFault function is done and returns.
+ * We also need to pass fault_va to the next breakpoint as it will be impossible to determine what it was after the function is executed.
+ * MmAccessFault is called if a allocated page is used for the first time.
+ * This method is based on some ideas of the IPT plugin
+
+ */
+static event_response_t mm_access_fault_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+
+    //Load the plugin object.
+    auto plugin = get_trap_plugin<hyperbee>(info);
+
+    //Could be used, if we trust the guest systems integrity
+    //Ignore any executables within System32 and SysWOW64
+    //    if (strstr(info->proc_data.name, "System32") != nullptr) {
+    //        return VMI_EVENT_RESPONSE_NONE;
+    //    }
+    //    if (strstr(dll_name_str, "SysWOW64") != nullptr) {
+    //         return VMI_EVENT_RESPONSE_NONE;
+    //    }
+
+    //Checks if a filter was set and applies it:
+    //This applies only to the trap set up. If the program dies in the meantime, the trap continues and might rise the
+    // callback even if another program is triggering it, since all depends only on the guest frame number.
+    if (plugin->hyperbee_filter_executable[0])
+    {
+        if (strcasestr(info->proc_data.name, plugin->hyperbee_filter_executable) == nullptr)
```

Comment:
```suggestion
        if (strcasestr(info->proc_data.name, plugin->hyperbee_filter_executable) == NULL)
```
`nullptr` is a C++ type, when calling C functions plese use `NULL`